plugins/generator-1.19.4/forge-1.19.4/templates/elementinits/villagertrades.java.ftl

```diff
@@ -0,0 +1,85 @@
+<#--
+ # MCreator (https://mcreator.net/)
+ # Copyright (C) 2012-2020, Pylo
+ # Copyright (C) 2020-2023, Pylo, opensource contributors
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # Additional permission for code generator templates (*.ftl files)
+ #
+ # As a special exception, you may create a larger work that contains part or
+ # all of the MCreator code generator templates (*.ftl files) and distribute
+ # that work under terms of your choice, so long as that work isn't itself a
+ # template for code generation. Alternatively, if you modify or redistribute
+ # the template itself, you may (at your option) remove this special exception,
+ # which will cause the template and the resulting code generator output files
+ # to be licensed under the GNU General Public License without this special
+ # exception.
+-->
+
+<#-- @formatter:off -->
+<#include "../mcitems.ftl">
+
+/*
+*	MCreator note: This file will be REGENERATED on each build.
+*/
+
+package ${package}.init;
+
+import net.minecraft.world.entity.npc.VillagerTrades;
+
+@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.FORGE) public class ${JavaModName}Trades {
+
+    <#if w.hasVillagerTrades(true)>
+    @SubscribeEvent public static void registerWanderingTrades(WandererTradesEvent event) {
+        <#list villagertrades as trade>
+            <#list trade.tradeEntries as tradeEntry>
+                <#if tradeEntry.villagerProfession == "WanderingTrader">
+                    <#list tradeEntry.entries as entry>
+                        event.getGenericTrades().add(
+                        new BasicItemListing(
+                        ${mappedMCItemToItemStackCode(entry.price1, entry.countPrice1)},
+                        <#if !entry.price2.isEmpty()>${mappedMCItemToItemStackCode(entry.price2, entry.countPrice2)},</#if>
+                        ${mappedMCItemToItemStackCode(entry.offer, entry.countOffer)},
+                        ${entry.maxTrades}, ${entry.xp}, ${entry.priceMultiplier}f
+                        ));
```

Comment:
Shouldn't it look like this?
```suggestion
                        event.getGenericTrades().add(
                            new BasicItemListing(
                                ${mappedMCItemToItemStackCode(entry.price1, entry.countPrice1)},
                                <#if !entry.price2.isEmpty()>${mappedMCItemToItemStackCode(entry.price2, entry.countPrice2)},</#if>
                                ${mappedMCItemToItemStackCode(entry.offer, entry.countOffer)},
                                ${entry.maxTrades}, ${entry.xp}, ${entry.priceMultiplier}f
                            )
                        );
```

Reply:
I just copied the 1.19.2 template... [https://github.com/MCreator/MCreator/blob/master/plugins/generator-1.19.2/forge-1.19.2/templates/elementinits/villagertrades.java.ftl](https://github.com/MCreator/MCreator/blob/master/plugins/generator-1.19.2/forge-1.19.2/templates/elementinits/villagertrades.java.ftl)
Apparently, no changes were needed

Reply:
Copying files in IDEA messes up indentations. Use system copy or fix indentations manually

Reply:
You don't get It, the indentations are the same as the 1.19.2 template

Reply:
I see. Does not mean that while porting, cleanup and indentation improvements shouldn't be done, if there are places found where it would be nice to have. Not mandatory, though, as long as indentations are kept at least the same as before

Reply:
It's done